src/ophyd_async/epics/motion/motor.py

```diff
@@ -93,5 +184,48 @@ async def set(
     async def stop(self, success=False):
         self._set_success = success
         # Put with completion will never complete as we are waiting for completion on
-        # the move above, so need to pass wait=False
+        # the move above, so need to pass wait=False.
         await self.motor_stop.trigger(wait=False)
+
+    async def _prepare_velocity(
+        self, start_position, end_position, time_for_move
+    ) -> float:
+        fly_velocity = (start_position - end_position) / time_for_move
+        max_speed = await self.max_velocity.get_value()
+        egu = await self.motor_egu.get_value()
+        if abs(fly_velocity) > max_speed:
+            raise MotorLimitsException(
+                f"Motor speed of {abs(fly_velocity)} {egu}/s was requested for a motor "
+                f" with max speed of {max_speed} {egu}/s"
+            )
+        await self.velocity.set(abs(fly_velocity))
+        return fly_velocity
+
+    async def _prepare_motor_path(
+        self, fly_velocity: float, start_position: float, end_position: float
+    ) -> float:
+        # Distance required for motor to accelerate from stationary to fly_velocity, and
+        # distance required for motor to decelerate from fly_velocity to stationary
+        run_up_distance = (await self.acceleration_time.get_value()) * fly_velocity
+
+        fly_prepared_position = start_position - run_up_distance
+
+        self._fly_completed_position = end_position + run_up_distance
+
```

Comment:
on the first look it's not clear why one variable is put into 'self' and another is not